libp2p/vbuffer.nim

```diff
@@ -24,9 +24,15 @@ template isEnough*(vb: VBuffer, length: int): bool =
   ## Returns ``true`` if buffer ``vb`` holds at least ``length`` bytes.
   len(vb.buffer) - vb.offset - length >= 0
 
+proc high*(vb: VBuffer): int =
+  ## Returns number of bytes left in buffer ``vb``.
+  result = vb.buffer.high - vb.offset
+  doAssert(result >= 0)
```

Comment:
I added those asserts indeed because that's how you write utility like this... asserting

Reply:
`high == len -1`, so what happens if `len == 0`?

Reply:
You are replacing all instances of `len(vbuffer.buffer) - 1` with this construction... its totally wrong.

Reply:
I really not sure about this helper procedure.

Reply:
it's needed in nim for `toOpenArray` because of the funny end indexing

Reply:
> `high == len -1`, so what happens if `len == 0`?

What's the behavior of nim `high` in such case? .. afaik `-1`

Reply:
indeed :) you can also implement it as `len - 1` then the assert isn't needed really